### PR TITLE
🐛 monitor middleware - fix ignore custom settings

### LINKED ISSUE #2019: 🐛 [Bug]: Monitor middleware ignores custom settings if default title is used.
### Bug Description

When using the monitor middleware I noticed that if I don't change the title or refresh rate the other custom settings don't get applied. 

The bug is on this line: https://github.com/gofiber/fiber/blob/master/middleware/monitor/config.go#L99

Where if title or refresh rate are the `Default Values` everything else gets ignored in the `else`.

### How to Reproduce

Run code

### Expected Behavior

Custom settings should be applied regardless of title/refresh rate.

### Fiber Version

v2.36.0

### Code Snippet (optional) (Bug, Font/Chart URLs are ignored)

```go
package main

import (
	"log"

	"github.com/gofiber/fiber/v2"
	"github.com/gofiber/fiber/v2/middleware/monitor"
)

func main() {
	app := fiber.New()

	app.Get("/metrics", monitor.New(monitor.Config{
	    FontURL: "blah/blah.css",
	    ChartJsURL: "blah/blah.js",
    }))

	log.Fatal(app.Listen(":3000"))
}
```

### Code Snippet (Working, Font/Chart URLs get applied)

```go
package main

import (
	"log"

	"github.com/gofiber/fiber/v2"
	"github.com/gofiber/fiber/v2/middleware/monitor"
)

func main() {
	app := fiber.New()

	app.Get("/metrics", monitor.New(monitor.Config{
	    Title: "MyService Metrics Page",
	    FontURL: "blah/blah.css",
	    ChartJsURL: "blah/blah.js",
    }))

	log.Fatal(app.Listen(":3000"))
}
```

This bug doesn't show up in monitor_test.go, because the test is changing the `Title` to "New + DefaultTitle".

### Checklist:

- [X] I agree to follow Fiber's [Code of Conduct](https://github.com/gofiber/fiber/blob/master/.github/CODE_OF_CONDUCT.md).
- [X] I have checked for existing issues that describe my problem prior to opening this one.
- [X] I understand that improperly formatted bug reports may be closed without explanation.

### PR DESCRIPTION
**Pull request by wangjq4214**
**Please provide enough information so that others can review your pull request:**

Fix monitor middleware ignores custom settings if default title is used.

**Explain the *details* for making this change. What existing problem does the pull request solve?**

close #2019

### CODE REVIEW

**Review by efectn — COMMENTED** [review]
Can you add some test cases

**Comment by ReneWerner87** [comment]
> Can you add some test cases

would be nice 
then we probably would have noticed it in the beginning and no one can break it without us noticing it

File: middleware/monitor/config.go [hunk 1]
```diff
@@ -115,5 +108,13 @@ func configDefault(config ...Config) Config {
 		cfg.APIOnly = ConfigDefault.APIOnly
 	}
 
+	if cfg.Title == ConfigDefault.Title && cfg.Refresh == ConfigDefault.Refresh {
+		cfg.index = ConfigDefault.index
+	} else {
+		// update cfg.index with custom title/refresh
+		cfg.index = newIndex(viewBag{cfg.Title,
+			cfg.Refresh, cfg.FontURL, cfg.ChartJsURL, cfg.CustomHead})
```
Comment (gaby) [inline middleware/monitor/config.go:116]: These changes will not get applied if the `IF` above is True.

**Review by gaby — CHANGES_REQUESTED** [review]

**Review by gaby — CHANGES_REQUESTED** [review]
These changes are not fixing the problem. The else statement never runs if the default `Title` and `Refresh` are used which is the original issue

**Comment by wangjq4214** [comment]
> These changes are not fixing the problem. The else statement never runs if the default `Title` and `Refresh` are used which is the original issue

To be honest I don't really understand what the point of this judgment is, it seems to me to be unnecessary.

**Comment by gaby** [comment]
> > These changes are not fixing the problem. The else statement never runs if the default `Title` and `Refresh` are used which is the original issue
> 
> To be honest I don't really understand what the point of this judgment is, it seems to me to be unnecessary.

No judging, just wanted to point the issue with the URL's. The original commit fixed the issue for the "refresh" field. Looks better now! 👍🏻

**Comment by wangjq4214** [comment]
@ReneWerner87 @efectn @gaby Please check out the new changes when you have time.

**Review by gaby — APPROVED** [review]
Looks good to me! Thanks! 🚀

**Review by ReneWerner87 — APPROVED** [review]

**Pull request merged**